jax/numpy/lax_numpy.py

```diff
@@ -3055,14 +3055,38 @@ def polyadd(a1, a2):
 def polyder(p, m=1):
   p = asarray(p)
   if m < 0:
-    raise ValueError("Order of derivative must be positive")
+    raise ValueError("Order of derivative must be nonnegative")
   if m == 0:
     return p
   if m % 1:
     raise ValueError("m must be an integer")
-  coeff = (arange(len(p), m, -1) - 1 - arange(m)[:, newaxis]).prod(0)
+  coeff = (np.arange(len(p), m, -1) - 1 - np.arange(m)[:, newaxis]).prod(0)
   return p[:-m] * coeff
 
+
+@_wraps(np.polyint)
+def polyint(p, m=1, k=None):
+  p = asarray(p)
```

Comment:
We're moving toward requiring inputs to be arrays to start with (though proceeding carefully for existing functions for backward compatibility). For a new function like this, we should call `_check_arraylike('polyint', p)` before `p = asarray(p)`.

Similar for `k` when it is not `None`.